x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/rules_table_filters/tags_filter_popover.tsx

```diff
@@ -62,8 +64,12 @@ const TagsFilterPopoverComponent = ({
   tags,
   selectedTags,
   onSelectedTagsChanged,
+  currentFilterTags,
 }: TagsFilterPopoverProps) => {
-  const sortedTags = useMemo(() => caseInsensitiveSort(tags), [tags]);
+  const sortedTags = useMemo(() => caseInsensitiveSort(union(tags, currentFilterTags)), [
```

Comment:
If not wanting to pull in `lodash` this is a nice and easy way to get uniques 🙂 

``` ts
Array.from(new Set(tags, currentFilterTags))
```

Reply:
👍  smort